async-http-client-backend/zio/src/main/scala/com/softwaremill/sttp/asynchttpclient/zio/AsyncHttpClientZioBackend.scala

```diff
@@ -13,29 +15,43 @@ import org.asynchttpclient.{
   DefaultAsyncHttpClientConfig
 }
 import org.reactivestreams.Publisher
+import io.netty.buffer.{ByteBuf, Unpooled}
 import scalaz.zio._
+import scalaz.zio.stream._
+import scalaz.zio.interop.reactiveStreams._
 
 class AsyncHttpClientZioBackend private (asyncHttpClient: AsyncHttpClient, closeClient: Boolean)
-    extends AsyncHttpClientBackend[IO[Throwable, ?], Nothing](asyncHttpClient, IOMonadAsyncError, closeClient) {
+    extends AsyncHttpClientBackend[Task, Stream[Throwable, ByteBuffer]](asyncHttpClient, IOMonadAsyncError, closeClient) {
 
-  override protected def streamBodyToPublisher(s: Nothing): Publisher[ByteBuf] =
-    s // nothing is everything
+  val runtime = new DefaultRuntime {}
 
-  override protected def publisherToStreamBody(p: Publisher[ByteBuffer]): Nothing =
-    throw new IllegalStateException("This backend does not support streaming")
+  private val bufferSize = 16
+
+  override protected def streamBodyToPublisher(s: Stream[Throwable, ByteBuffer]): Publisher[ByteBuf] =
+    runtime.unsafeRunSync(s.map(Unpooled.wrappedBuffer).toPublisher).toEither.right.get
```

Comment:
`runtime.unsafeRunSync(...).toEither.right.get` can be simplified to `runtime.unsafeRun(...)`